homeassistant/components/heos/media_player.py

```diff
@@ -122,12 +136,66 @@ async def async_added_to_hass(self):
                 SIGNAL_HEOS_UPDATED, self._heos_updated
             )
         )
+        self._group = await self.async_get_group()
 
     @log_command_error("clear playlist")
     async def async_clear_playlist(self):
         """Clear players playlist."""
         await self._player.clear_queue()
 
+    async def async_get_group(self):
+        """Build a dictionary with HEOS group membership information."""
+        group_membership = []
+        if not self.entity_id:
+            return group_membership
+        if not self.player_id:
+            _LOGGER.info(
+                "No player_id for %s yet, not returning group info.", self.entity_id
+            )
+            return group_membership
+
+        controller = self.hass.data[HEOS_DOMAIN][DATA_CONTROLLER_MANAGER].controller
+        if controller.connection_state != heos_const.STATE_CONNECTED:
+            _LOGGER.error("Unable to rebuild group because HEOS is not connected")
+            return group_membership
+
+        try:
+            groups = await controller.get_groups(refresh=True)
+        except HeosError as err:
+            _LOGGER.error("HEOS Unable to get group info: %s", err)
+            return group_membership
+
+        for group in groups.values():
+            member_player_ids = {member.player_id for member in group.members}
+            if (
+                self.player_id == group.leader.player_id
+                or self.player_id in member_player_ids
+            ):
+                leader_entity_id = None
+                for entity in self.hass.data[DOMAIN].entities:
+                    if not isinstance(entity, HeosMediaPlayer):
+                        continue
+                    if entity.player_id == group.leader.player_id:
+                        leader_entity_id = entity.entity_id
+                    elif entity.player_id in member_player_ids:
+                        group_membership.append(entity.entity_id)
+                if not leader_entity_id:
```

Comment:
Why would this ever occur?

Reply:
In my tests it has never occured, so I have removed the check.